Enhancing UX for adding content

Dante291 opened the pull request:
**What kind of change does this PR introduce?**

Enhancing UX for adding content so that no longer add sign button are there in a particular screen of the App.

**Issue Number:**

Fixes #2191 

**Did you add tests for your changes?**

Yes 

**Snapshots/Videos:**
![Screenshot_1701688091](https://github.com/PalisadoesFoundation/talawa/assets/103507835/189abbeb-fd15-44b7-88b1-b29210be188d)


**Summary**

Currently there are more than one add/plus sign button on some screen of the app to add content this affects the user experience of the App. I added Floating Action Button(FAB) on the feed screen of the app so that there wouldn't be more than one add/plus button on any screen of the app.

This FAB navigates us to the add post page screen to make new posts, took reference from X/Twitter app for making new tweets.


**Have you read the [contributing guide] (https://github.com/PalisadoesFoundation/talawa/blob/master/CONTRIBUTING.md)?**

Yes

literalEval commented on lib/constants/routing_constants.dart:
```diff
@@ -99,4 +99,7 @@ class Routes {
   /// static variable to access pinnedpostscreen.
   static const String pinnedPostScreen = '/pinnedpostscreen';
+
+  /// static variable.
```
Write a better doc.

literalEval commented on lib/views/after_auth_screens/feed/organization_feed.dart:
```diff
@@ -26,6 +27,19 @@ class OrganizationFeed extends StatelessWidget {
       builder: (context, model, child) {
         print(model.posts);
```
Can you remove this print statement as a part of your PR itself ?

literalEval commented on lib/views/after_auth_screens/feed/organization_feed.dart:
```diff
@@ -26,6 +27,19 @@ class OrganizationFeed extends StatelessWidget {
+            },
+            child: const Icon(
+              Icons.add,
+              size: 35,
```
Don't use constant sizing. Use SizeConfig instead.

literalEval commented:
@Dante291 since it is a UI change, it'll help if you can please add before and after screen records.

Ayush0Chaudhary commented:
@Dante291 add code coverge ss too

Ayush0Chaudhary reviewed (commented):
Make sure you cover all the lines.

Dante291 commented:
![Screenshot_2023-12-05-23-47-44-236_com example talawa](https://github.com/PalisadoesFoundation/talawa/assets/103507835/40dd0a35-8110-43dc-a80a-971878de7cfa)

Dante291 commented:
![image](https://github.com/PalisadoesFoundation/talawa/assets/103507835/30bb8e08-5a2b-4c4c-ab60-6e3f47fe2681)


Added test for all the lines I have added/edited.

Dante291 commented:
> Don't use constant sizing. Use SizeConfig instead.

@literalEval I tried using SizeConfig for setting up the size of the icon but it showed me error, Invalid constant value, I also noticed constant sizing is used for setting up the size for Icons throughout the app.

Other than this I have resolved all the above asked changes.

noman2002 commented:
> > Don't use constant sizing. Use SizeConfig instead.
> 
> @literalEval I tried using SizeConfig for setting up the size of the icon but it showed me error, Invalid constant value, I also noticed constant sizing is used for setting up the size for Icons throughout the app.
> 
> Other than this I have resolved all the above asked changes.

Remove const from the widget. You'll not the error.

Dante291 commented:
> > > Don't use constant sizing. Use SizeConfig instead.
> > 
> > 
> > @literalEval I tried using SizeConfig for setting up the size of the icon but it showed me error, Invalid constant value, I also noticed constant sizing is used for setting up the size for Icons throughout the app.
> > Other than this I have resolved all the above asked changes.
> 
> Remove const from the widget. You'll not get the error.

@noman2002  I didn't notice that, Thanks sir it helped. Please review or if you have any other UI or any changes do lemme know.
